test/functional/utils/create-connection.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Connection,
+  ConnectionOptions,
+  createConnection as typeormCreateConnection,
+} from 'typeorm'
+
+export const createConnection = async (partialOptions: Partial<ConnectionOptions> = {}) => {
+  return typeormCreateConnection({
+    ...partialOptions,
+    type: 'aurora-data-api',
+    database: process.env.database!,
+    secretArn: process.env.secretArn!,
+    resourceArn: process.env.resourceArn!,
+    region: process.env.region!,
+    logging: true,
+    logger: 'simple-console',
+    extra: {
+      httpOptions: {
+        connectTimeout: 120000,
+      },
+      maxRetries: 50,
+      retryDelayOptions: {
+        base: 10000,
+      },
+      ...(partialOptions.extra || {}),
+    },
+  })
+}
+
+export const createConnectionAndResetData = async (
+  partialOptions: Partial<ConnectionOptions> = {},
+) => {
+  // FIXME: Whatever happens here the data is not dropped!
+  // Tried: await connection.synchronize(true)
+  // Tried: await connection.dropDatase()
+  // And the following synchronize/dropSchema combination
+  // Bizarre - so for now deleting the data in the tests as need to move on....
+  return createConnection({ ...partialOptions, synchronize: true, dropSchema: true })
+}
+
+export const useCleanDatabase = async (
+  partialOptions: Partial<ConnectionOptions> = {},
+  invoke: (connection: Connection) => Promise<void>,
+) => {
+  const connection = await createConnectionAndResetData(partialOptions)
+  try {
+    await invoke(connection)
+  } finally {
+    await connection.close()
```

Comment:
Guess we can run `drop table all` here to clean the db before closing the connection to fix the issue described above.
I think we might want to make two functions out of that and just run them `afterEach` or `beforeEach`. That will allow us to not do the callback.

Reply:
Sure I will do this and I will drop the callback however I quite like that approach as it means there's no before and after functions, less overall code, and no duplication of test method setup/teardown code in other test classes.

But no problem will remove.

Reply:
Plus no mutating module scoped variables to hold the connection. But nearly changed.

Reply:
If it's not too late, you can leave it like that. I was just throwing ideas here, it might be nicer than `afterEach` and `beforeEach`

Reply:
I haven't done this just yet as when I went to implement it we still would have to create the connection in the method as we need to pass in the correct options (primarily entities). This means although we could set a module scoped connection variable and then clean it up in afterEach that feels a bit smelly to me as it introduces a mutating connection variable and makes the code asymmetrical. 

If you have a strong opinion here let me know and I can fix it or you can make adjustments perhaps later?

Reply:
Your reasoning makes sense, let's leave it like that.